force-app/src/classes/Argument.cls

```diff
@@ -144,57 +148,62 @@ global class Argument {
       this.jsonValue = JSON.serialize(callArgumentToMatch);
     }
 
-    public boolean matches(final Object callArgument) {
-      return this.jsonValue == JSON.serialize(callArgument);
+    public Boolean matches(final Object callArgument) {
+      if (this.jsonValue == null) {
+        return callArgument == null;
+      }
+
+      return this.jsonValue.equals(JSON.serialize(callArgument));
     }
```

Comment:
**suggestion:** `JSON.serialize()` never returns `null`, it returns `'null'``

```apex
Object o = null;
System.Debug(JSON.serialize(o) == null); // => displays "false"
System.Debug(JSON.serialize(o) == 'null'); // => displays "true"
```

So it means this check will never be true.
Then we don't need this check and we can return `this.jsonValue.equals(JSON.serialize(callArgument));`

```suggestion
    public Boolean matches(final Object callArgument) {
      return this.jsonValue.equals(JSON.serialize(callArgument));
    }
```